File: Data Structures & Algorithms/meeting-schedule/submission-2.java
/**
 * Definition of Interval:
 * public class Interval {
 *     public int start, end;
 *     public Interval(int start, int end) {
 *         this.start = start;
 *         this.end = end;
 *     }
 * }
 */


class Solution {
    public boolean canAttendMeetings(List<Interval> intervals) {

        Collections.sort(intervals, Comparator.comparingInt(i -> i.start));

        List<Integer> intList = new ArrayList<>();

        for (Interval interval : intervals) {
            intList.add(interval.start);  
            intList.add(interval.end);    
        }

        for (int i = 0; i < intList.size() - 1; i++) {
            if (intList.get(i + 1) < intList.get(i)) {
                return false;
            }
        }

        return true;
    }
}

File: Data Structures & Algorithms/non-overlapping-intervals/submission-2.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        
        intervals.sort()
        prevEnd = intervals[0][1]
        res = 0

        for start, end in intervals[1:]:
            # No overlapping
            if start >= prevEnd:
                prevEnd = end
            else:
                res += 1
                # Delete interval with "greated" end
                prevEnd = min(prevEnd, end)
        return res